sql/analyzer/resolve_columns.go

```diff
@@ -1010,9 +1047,14 @@ func pushdownGroupByAliases(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Sc
 }
 
 // replaceExpressionsWithAliases replaces any expressions in the slice given that match the map of aliases given with
-// their alias expression. This is necessary when pushing aliases down the tree, since we introduce a projection node
-// that effectively erases the original columns of a table.
+// their alias expression or alias name. This is necessary when pushing aliases down the tree, since we introduce a
+// projection node that effectively erases the original columns of a table.
 func replaceExpressionsWithAliases(exprs []sql.Expression, replacedAliases map[string]string) ([]sql.Expression, transform.TreeIdentity) {
+	newMap := make(map[string]struct{})
```

Comment:
This doesn't seem to be necessary, it's always an exact copy of replacedAliases and you're not writing to either map after init

Reply:
whoops... `newMap` is an embarrassing name! ;-) Thanks for poking on this; I meant to clean that temporary name up, but it snuck through. 

I renamed it to `replacedAliasesByName` and renamed in the input parameter from `replacedAliases` to `replacedAliasesByExpression`. That naming should make the intent much more obvious now. It's a reverse lookup map for the replaced aliases, so we can check for any unqualified columns that use that alias and tag them as an `AliasReference`. It's needed so that `qualifyColumns` sees those are AliasReferences and won't try to qualify them to a table when the name matches a column name.